packages/core/src/external/ehr/command/get-appointments/ehr-get-appointments.ts

```diff
@@ -0,0 +1,23 @@
+import { SlimBookedAppointment } from "@metriport/shared/interface/external/ehr/canvas/appointment";
+
+export type GetAppointmentsRequest = {
+  method: AppointmentMethods;
+  tokenId?: string;
+  cxId: string;
+  practiceId: string;
+  departmentIds?: string[];
+  fromDate?: Date;
+  toDate?: Date;
+};
+
+export type GetAppointmentsClientRequest = Omit<GetAppointmentsRequest, "method">;
+
+export interface EhrGetAppointmentsHandler {
+  getAppointments<T extends Appointment>(request: GetAppointmentsRequest): Promise<T[]>;
+}
+
+export type Appointment = SlimBookedAppointment;
+
+export enum AppointmentMethods {
+  canvasGetAppointments = "canvasGetAppointments",
+}
```

Comment:
I thought I added a comment about this on the previous PR. The problem with this type of code is that there's no compilation-time assessment that we're providing the expected parameters.

My suggestion is to do something like this:

<img width="1575" alt="image" src="https://github.com/user-attachments/assets/b00835c5-b15c-4e15-96f8-d600a1e56e47" />

<img width="1580" alt="image" src="https://github.com/user-attachments/assets/7e6ffc59-87c7-4d29-bca9-143415d3ffb7" />

<img width="1529" alt="image" src="https://github.com/user-attachments/assets/d6666966-823e-4b82-ac22-b9f1247d9440" />